src/main/java/ezschedule/model/event/Event.java

```diff
@@ -61,21 +68,50 @@ public boolean equals(Object other) {
 
         Event otherEvent = (Event) other;
         return otherEvent.getName().equals(getName())
-            && otherEvent.getDate().equals(getDate())
-            && otherEvent.getStartTime().equals(getStartTime())
-            && otherEvent.getEndTime().equals(getEndTime());
+                && otherEvent.getDate().equals(getDate())
+                && otherEvent.getStartTime().equals(getStartTime())
+                && otherEvent.getEndTime().equals(getEndTime());
     }
 
     @Override
     public String toString() {
         final StringBuilder sb = new StringBuilder();
         sb.append(getName())
-            .append("\nDate: ")
-            .append(getDate())
-            .append("\nStart Time: ")
-            .append(getStartTime())
-            .append("\nEnd End: ")
-            .append(getEndTime());
+                .append("\nDate: ")
+                .append(getDate())
+                .append("\nStart Time: ")
+                .append(getStartTime())
+                .append("\nEnd End: ")
+                .append(getEndTime());
         return sb.toString();
     }
+
+    private boolean isEqualDate(Event otherEvent) {
+        return this.getDate().equals(otherEvent.getDate());
+    }
+
+    private boolean isTimeOverlap(Event otherEvent) {
+        return isStartTimeOverlap(otherEvent) || isEndTimeOverlap(otherEvent)
+                || isTimeInBetween(otherEvent) || isTimeIsEqual(otherEvent);
+    }
+
+    private boolean isStartTimeOverlap(Event otherEvent) {
+        return otherEvent.getStartTime().isBefore(this.getStartTime())
+                && otherEvent.getEndTime().isAfter(this.getStartTime());
+    }
+
+    private boolean isEndTimeOverlap(Event otherEvent) {
```

Comment:
Is `isStartTimeOverlap` the same as `isEndTimeOverlap`?

Reply:
Nope. My idea was that we have to check if the otherEvent either overlaps in start time or end time. Should either one be overlap, it could not be added.